cdap-ui/app/services/data/datasource.js

```diff
@@ -148,27 +167,24 @@ angular.module(PKG.name+'.services')
     };
 
     /**
-     * fetch a resource
+     * Fetch a resource on-demand. Send the action 'request' to 
+     * the node backend.
      */
     DataSource.prototype.request = function (resource, cb) {
-      var once = false,
-          deferred = $q.defer();
+      var deferred = $q.defer();
 
+      var id = generateUUID();
+      resource.id = id;
```

Comment:
We use it to determine which callback to execute when the response comes from the node server (line 93)